subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/IndexedNormalizedFileSnapshot.java

```diff
@@ -49,8 +49,9 @@ protected boolean hasSameNormalizedPathAs(AbstractNormalizedFileSnapshot other)
                 return false;
             }
             return absolutePath.regionMatches(index, that.absolutePath, that.index, myLength);
+        } else {
+            return super.hasSameNormalizedPathAs(other);
```

Comment:
Was this a bug?

Reply:
@lptr that's the job of the equals method, not this method.

@wolfs No, it doesn't matter because the equals method was the only client of this one, I just changed it for consistency.

Reply:
@lptr The normalization is not necessarily different if we use a different implementation to store the normalized path. For example, an `IndexedNormalizedFileSnapshot` can denote the same normalized path as `DefaultNormalizedFileSnapshot`. I am somewhat tempted to say that the `equals` method should not compare the actual classes. WDYT?

Reply:
This is just a theoretical possibility. We should not have different types compared to each other within the same machine. If we do, that's a bug we should fix.